controllers/gatekeeper_controller.go

```diff
@@ -186,3 +242,59 @@ func (r *GatekeeperReconciler) updateOrCreateResource(manifest *manifest.Manifes
 
 	return err
 }
+
+func (r *GatekeeperReconciler) finalizeGatekeeper(reqLogger logr.Logger, gatekeeper *operatorv1alpha1.Gatekeeper) error {
+	ctx := context.Background()
+
+	var err error
+	for _, a := range orderedStaticAssets {
+		assetName := staticAssetsDir + a
+		bytes, err := bindata.Asset(assetName)
+		if err != nil {
+			return errors.Wrapf(err, "Unable to retrieve bindata asset %s", assetName)
+		}
+
+		manifest := &manifest.Manifest{}
+		err = manifest.UnmarshalJSON(bytes)
+		if err != nil {
+			return errors.Wrapf(err, "Unable to unmarshal YAML bytes for asset name %s", assetName)
+		}
+
+		// Delete cluster scoped resource not owned by the CR
+		if manifest.Obj.GetNamespace() == "" {
+
+			err = r.Delete(ctx, manifest.Obj)
+			if err != nil && !apierrors.IsNotFound(err) {
+				return errors.Wrapf(err, "Error Deleting Finalizer Resource. Kind: '%s'. Name: '%s'", manifest.GVK.Kind, manifest.Obj.GetName())
+			}
+
+		}
+
+	}
+
+	reqLogger.Info("Successfully finalized Gatekeeper")
+	return err
+}
+
+func (r *GatekeeperReconciler) addFinalizer(reqLogger logr.Logger, g *operatorv1alpha1.Gatekeeper) error {
+	ctx := context.Background()
+
+	controllerutil.AddFinalizer(g, gatekeeperFinalizer)
+
+	// Update CR
+	err := r.Update(ctx, g)
+	if err != nil {
+		reqLogger.Error(err, "Failed to update Gatekeeper with finalizer")
+		return err
+	}
+	return nil
+}
+
+func contains(list []string, s string) bool {
```

Comment:
We can simplify this by using the apimachinery sets package https://godoc.org/k8s.io/apimachinery/pkg/util/sets#String. We just create a new string set `NewString()` containing the finalizers from `GetFinalizers()` and then invoke the `Has()` API method.